File: app/src/main/java/net/veldor/flibustaloader/database/entity/BooksDownloadSchedule.java
package net.veldor.flibustaloader.database.entity;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import org.jetbrains.annotations.NotNull;

@Entity
public class BooksDownloadSchedule{
    @PrimaryKey(autoGenerate = true)
    public int id;

    @NotNull
    public String bookId = "";
    @NotNull
    public String link = "";
    @NotNull
    public String name = "";
    @NotNull
    public String size = "";
    @NotNull
    public String author = "";
    @NotNull
    public String format = "";

    public String authorDirName = "";

    public String sequenceDirName = "";

    public String reservedSequenceName = "";

    @Ignore
    public boolean loaded;
}

File: app/src/main/java/net/veldor/flibustaloader/database/entity/BooksDownloadSchedule.kt
package net.veldor.flibustaloader.database.entity

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import net.veldor.flibustaloader.selections.CurrentBookDownloadProgress

@Entity
class BooksDownloadSchedule {


    @Ignore
    var progress: CurrentBookDownloadProgress? = null

    @PrimaryKey(autoGenerate = true)
    var id = 0
    var bookId = ""
    @JvmField
    var link = ""
    @JvmField
    var name = ""
    var size = ""
    var author = ""
    @JvmField
    var format = ""
    @JvmField
    var authorDirName = ""
    @JvmField
    var sequenceDirName = ""
    @JvmField
    var reservedSequenceName = ""

    @JvmField
    @Ignore
    var loaded = false

    @JvmField
    @Ignore
    var failed: Boolean = false

    @JvmField
    @Ignore
    var inProgress: Boolean = false
}